core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -400,7 +407,8 @@ class ControllerApis(val requestChannel: RequestChannel,
         if (!authorizedTopicNames.contains(name)) {
           response.topics().add(new CreatableTopicResult().
             setName(name).
-            setErrorCode(TOPIC_AUTHORIZATION_FAILED.code))
+            setErrorCode(TOPIC_AUTHORIZATION_FAILED.code).
+            setErrorMessage("Authorization failed."))
```

Comment:
Why add this?

Reply:
@hachikuji Mainly because the ZK implementation returns this error message. For the test that I added in this PR, I want it to succeed in both mode (ZK and KRaft). For this, I need to either add this error message to the KRaft impl or remove this message from the ZK impl. This seems like the more backward compatible change.

Reply:
Ok. I thought we left the error message empty whenever the message did not have additional detail over the default message in `Errors`. I think it would be fine to remove the message in both cases as well. I'll leave it up to you.

Reply:
Thanks for the background information. I found another issue while trying to implement this. The default for `CreateTopicsResponse.errorMessage` is the empty string, `""`, not `null`. This means that `ApiError.messageWithFallback` doesn't behave as expected.

I am going to create an issue for this and tackle it in another PR.